pkgs/applications/networking/instant-messengers/discord/openasar.nix

```diff
@@ -0,0 +1,63 @@
+{ lib, stdenv, fetchFromGitHub, fetchYarnDeps, nodejs-16_x, yarn, bash, fixup_yarn_lock }:
+
+stdenv.mkDerivation rec {
+  version = "unstable-2022-06-10";
+  pname = "openasar";
+
+  src = fetchFromGitHub {
+    owner = "GooseMod";
+    repo = "OpenAsar";
+    rev = "c6f2f5eb7827fea14cb4c54345af8ff6858c633a";
+    sha256 = "m6e/WKGgkR8vjKcHSNdWE25MmDQM1Z3kgB24OJgbw/w=";
+  };
+
+  nativeBuildInputs = [ nodejs-16_x yarn bash fixup_yarn_lock ];
```

Comment:
Can't you use something like `yarn2nix` instead of making your own build system :thinking: ?